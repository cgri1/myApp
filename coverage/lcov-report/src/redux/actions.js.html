<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/redux/actions.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">src/redux</a> actions.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/4</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { store } from "./store";
&nbsp;
export function <span class="fstat-no" title="function not covered" >setData(</span>data) {
<span class="cstat-no" title="statement not covered" >  store.dispatch({</span>
    type: "SET_DATA",
    payload: {
      data
    }
  });
}
&nbsp;
&nbsp;
&nbsp;
export function <span class="fstat-no" title="function not covered" >setStoryList(</span>data) {
<span class="cstat-no" title="statement not covered" >  store.dispatch({</span>
    type: "SET_STORY_LIST",
    payload: {
      data
    }
  });
}
&nbsp;
export function <span class="fstat-no" title="function not covered" >setVoteStatus(</span>data) {
<span class="cstat-no" title="statement not covered" >  store.dispatch({</span>
    type: "SET_VOTE_STATUS",
    payload: {
      data
    }
  });
}
&nbsp;
export function <span class="fstat-no" title="function not covered" >setRoute(</span>flag) {
<span class="cstat-no" title="statement not covered" >  store.dispatch({</span>
    type: "SET_ROUTE",
    payload: {
      flag
    }
  });
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Mar 23 2020 10:17:10 GMT+0300 (GMT+03:00)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
