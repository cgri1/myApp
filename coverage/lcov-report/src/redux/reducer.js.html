<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/redux/reducer.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">src/redux</a> reducer.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">55.56% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">28.57% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">55.56% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/9</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import assign from "lodash/assign";
&nbsp;
const defaultValues = {
  data: [],
  storyList:[],
  votedList: [],
  flag: false
};
&nbsp;
export const setData = function (
  state = defaultValues,
  action = <span class="branch-0 cbranch-no" title="branch not covered" >{ type: "UNKNOWN" }</span>
) {
  const { payload } = action;
  switch (action.type) {
<span class="branch-0 cbranch-no" title="branch not covered" >    case "SET_DATA":</span>
<span class="cstat-no" title="statement not covered" >      return assign({}, state, {</span>
        data: payload.data
      });
<span class="branch-1 cbranch-no" title="branch not covered" >    case "SET_STORY_LIST":</span>
<span class="cstat-no" title="statement not covered" >      return assign({}, state, {</span>
        storyList: payload.data
      });
<span class="branch-2 cbranch-no" title="branch not covered" >    case "SET_VOTE_STATUS":</span>
<span class="cstat-no" title="statement not covered" >      return assign({}, state, {</span>
        votedList: payload.data
      });
<span class="branch-3 cbranch-no" title="branch not covered" >    case "SET_ROUTE":</span>
<span class="cstat-no" title="statement not covered" >      return assign({}, state, {</span>
        flag: payload.flag
      });
    default:
      return state;
  }
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Mar 23 2020 10:17:10 GMT+0300 (GMT+03:00)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
