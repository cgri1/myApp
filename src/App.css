.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
  pointer-events: none;
}

.App-header {
  background-color: #282c34;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(5px + 2vmin);
  color: white;
  padding-top: 60px;
}

.Detail-header {
  background-color: #282c34;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(5px + 2vmin);
  color: white;
  padding-top: 160px;
  padding-bottom: 390px;
}

.App-link {
  color: #61dafb;
}

.textField {
  margin-top: 100px;
}

