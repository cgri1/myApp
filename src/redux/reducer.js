import assign from "lodash/assign";

const defaultValues = {
  data: [],
  storyList:[],
  votedList: [],
  flag: false
};

export const setData = function (
  state = defaultValues,
  action = { type: "UNKNOWN" }
) {
  const { payload } = action;
  switch (action.type) {
    case "SET_DATA":
      return assign({}, state, {
        data: payload.data
      });
    case "SET_STORY_LIST":
      return assign({}, state, {
        storyList: payload.data
      });
    case "SET_VOTE_STATUS":
      return assign({}, state, {
        votedList: payload.data
      });
    case "SET_ROUTE":
      return assign({}, state, {
        flag: payload.flag
      });
    default:
      return state;
  }
};
