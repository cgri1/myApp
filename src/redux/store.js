import { createStore } from "redux";
import { setData } from "./reducer";
import { combineReducers } from "redux";

const reducercombined = combineReducers({ setData });

export const store = createStore(reducercombined);
