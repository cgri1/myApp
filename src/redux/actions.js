import { store } from "./store";

export function setData(data) {
  store.dispatch({
    type: "SET_DATA",
    payload: {
      data
    }
  });
}



export function setStoryList(data) {
  store.dispatch({
    type: "SET_STORY_LIST",
    payload: {
      data
    }
  });
}

export function setVoteStatus(data) {
  store.dispatch({
    type: "SET_VOTE_STATUS",
    payload: {
      data
    }
  });
}

export function setRoute(flag) {
  store.dispatch({
    type: "SET_ROUTE",
    payload: {
      flag
    }
  });
}
