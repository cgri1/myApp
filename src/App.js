import React, { Component } from "react";
import { connect } from "react-redux";
import "./App.css";
import AppBar from '@material-ui/core/AppBar';
import TextField from "@material-ui/core/TextField";
import TextareaAutosize from '@material-ui/core/TextareaAutosize';
import Button from '@material-ui/core/Button';
import { withRouter } from "react-router";
import * as data from "./redux/actions";
class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      sessionName: "",
      numberOfVoter: 0,
      storyList: ""
    }

    this.addStory = this.addStory.bind(this)
    this.setSessionName = this.setSessionName.bind(this)
    this.setNumberOfVote = this.setNumberOfVote.bind(this)
    this.startSession = this.startSession.bind(this)

  }

  addStory(event) {
    this.setState({
      storyList: event.target.value
    })
  }

  setSessionName(event) {
    if (event.target.value.length < 200) {
      this.setState({
        sessionName: event.target.value
      })
    }
  }

  setNumberOfVote(event) {
    if (parseInt(event.target.value) !== 0) {
      this.setState({
        numberOfVoter: event.target.value
      })
    }
  }

  startSession() {
    if (this.state.numberOfVoter !== 0 && this.state.sessionName !== "") {
      this.props.history.push('/scrummaster')
      let arrayOfData = []
      arrayOfData.push(this.state.sessionName)
      arrayOfData.push(this.state.numberOfVoter)
      arrayOfData.push(this.state.storyList)
      data.setData(arrayOfData)

      //initial vote status
      let arrayOfVoteStatus = []
      for (let x = 1; x <= this.state.numberOfVoter; x++) {
        this.arrayOfVoter = {}
        this.arrayOfVoter.name = "Voter " + x + " : "
        this.arrayOfVoter.status = "Not Voted"
        this.arrayOfVoter.point = "0"
        arrayOfVoteStatus.push(this.arrayOfVoter)
      }
      this.arrayOfVoter = {}
      this.arrayOfVoter.name = "Scrum Master : "
      this.arrayOfVoter.status = "Not Voted"
      this.arrayOfVoter.point = "0"
      arrayOfVoteStatus.push(this.arrayOfVoter)
      data.setVoteStatus(arrayOfVoteStatus)

      //initial storylist status
      this.storyList = this.state.storyList.split("\n")
      let arrayOfStoryStatus = []
      for (let x = 0; x < this.storyList.length; x++) {
        this.arrayOfVoter = {}
        this.arrayOfVoter.name = this.storyList[x]
        this.arrayOfVoter.point = " "
        this.arrayOfVoter.status = "Not Voted"
        arrayOfStoryStatus.push(this.arrayOfVoter)
      }
      data.setStoryList(arrayOfStoryStatus)

      fetch('http://localhost:8080/setData', {
        method: 'POST',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          sessionName: this.state.sessionName,
          numberOfVoter: this.state.numberOfVoter,
          storyList: arrayOfStoryStatus,
          voteList: arrayOfVoteStatus
        }),
      }).then((r) => {
        return r.json();
      }).then((rsp) => {
        if (rsp.success) {
          console.log("Login Success");
        } else {
          console.log(rsp.message);
        }
      }).catch((e) => {
        console.log("Login Failure");
      });
    }
  }

  render() {
    return (
      <div className="App">
        <AppBar
          style={{ backgroundColor: "#282c34" }}
          position="static"
        > SCRUM POKER </AppBar>

        <div className="textField">
          <TextField id="outlined-basic" label="Session Name" variant="outlined" onChange={this.setSessionName} />
          <TextField id="outlined-basic" type="number" label="Number of Voters" variant="outlined" style={{ marginLeft: "300px" }} onChange={this.setNumberOfVote} />
        </div>
        <div >
          <TextareaAutosize style={{ width: "740px", height: "300px", marginTop: "50px", marginBottom: "30px" }} placeholder="Paste your story list" rowsMax={20} onChange={this.addStory} />
        </div>
        <Button variant="contained" onClick={this.startSession}>Start Session</Button>
      </div>
    );
  }
}

const mapStateToProps = (state, ownProps) => ({
  arrayOfData: state.setData.data
});

App = withRouter(connect(mapStateToProps)(App));
export default App;
