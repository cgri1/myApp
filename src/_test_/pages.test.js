import React from "react";
import Pages from "../pages";
import ReactDOM from 'react-dom';
import { shallow, mount } from "enzyme";
import { Provider } from "react-redux";
import { store } from "../redux/store";
import { MemoryRouter } from 'react-router';
import App from "../App"
import ScrumMasterPage from "../components/scrummaster"
import Developer from "../components/developer.js";

test("Page snapshot", () => {
  const component = shallow(
    <Provider store={store}>
      <Pages />
    </Provider>
  );
  expect(component).toMatchSnapshot();
});


test('ScrumMasterPage path', () => {
  const wrapper = shallow(
    <MemoryRouter initialEntries={['/scrummaster']}>
      <Provider store={store}>
        <ScrumMasterPage />
      </Provider>
    </MemoryRouter>
  );
  expect(wrapper.find(ScrumMasterPage)).toHaveLength(1);
});

test('App path ', () => {
  const wrapper = shallow(
    <MemoryRouter initialEntries={['/']}>
      <Provider store={store}>
        <App />
      </Provider>
    </MemoryRouter>
  );
  expect(wrapper.find(App)).toHaveLength(1);
});

test('Developer path ', () => {
  const wrapper = shallow(
    <MemoryRouter initialEntries={['/developer']}>
      <Provider store={store}>
        <Developer />
      </Provider>
    </MemoryRouter>
  );
  expect(wrapper.find(Developer)).toHaveLength(1);
});