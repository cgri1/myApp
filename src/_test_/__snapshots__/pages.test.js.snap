// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "storeState": Object {
        "setData": Object {
          "data": Array [],
          "flag": false,
          "storyList": Array [],
          "votedList": Array [],
        },
      },
    }
  }
>
  <Pages />
</ContextProvider>
`;

exports[`Page snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "storeState": Object {
        "setData": Object {
          "data": Array [],
          "flag": false,
          "storyList": Array [],
          "votedList": Array [],
        },
      },
    }
  }
>
  <Pages />
</ContextProvider>
`;
